### PR TITLE
[NEW] Blockstack as decentralized auth provider

### PR DESCRIPTION
**Pull request by rodrigok**
### Electron / Desktop
This implementation doesn't work with our electron/desktop yet

### Admin / Service Settings

A number of admin settings are added for the package behaviour and auth request details.

- `Enable`
- `Generate username` - If true will generate the username automatically from the user's profile when no username came from blockstack, otherwise user will see the username selection screen
- `Auth description` - Description provided in auth request screen within Blockstack
- `Button label text` - Text of the login button on login screen

### Avatars

Avatars are loaded directly from Blockstack.org when user logins for the first time and saved as a local avatar, users can update the avatars from the profile page where we added the blockstack as one of the avatar source of suggestions.

### The no email thing

Because Blockstack accounts do not contain an email address, yet Rocket.Chat fundamentally requires one, I've found a workaround by populating the email field with their ID @ `blockstack.email`.

### Added dependencies

Rocket.Chat required `meteor-node-stubs` package to support the `crypto` module in client.

Replaces #10462
Closes #10462

image

localhost_3000_admin_blockstack laptop with hidpi screen

### CODE REVIEW

**Comment by sampaiodiego** [comment]
I'm a bit worried about `meteor-node-stub`, see https://github.com/meteor/node-stubs/issues/6

File: packages/rocketchat-blockstack/server/routes.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Meteor } from 'meteor/meteor';
+import { WebApp } from 'meteor/webapp';
+import { RocketChat } from 'meteor/rocketchat:lib';
+
+WebApp.connectHandlers.use('/_blockstack/manifest', Meteor.bindEnvironment(function(req, res) {
+	const name = RocketChat.settings.get('Site_Name');
+	const startUrl = Meteor.absoluteUrl();
+	const description = RocketChat.settings.get('Blockstack_Auth_Description');
+	const iconUrl = Meteor.absoluteUrl('assets/favicon_192');
```
Comment (sampaiodiego) [inline packages/rocketchat-blockstack/server/routes.js]: we should have a proper way to get assets' addresses.. [we do differently](https://github.com/RocketChat/Rocket.Chat/blob/347a26204ced260db37f21b7e6f8886323457498/packages/rocketchat-lib/server/lib/PushNotification.js#L27) on push notifications

File: packages/rocketchat-blockstack/server/main.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { Meteor } from 'meteor/meteor';
+
+import './routes.js';
+import './settings.js';
+import './loginHandler.js';
+
+Meteor.isDevelopment = (process.env.ROOT_URL.indexOf('localhost') !== -1);
```
Comment (sampaiodiego) [inline packages/rocketchat-blockstack/server/main.js]: this is not being used by this PR

Comment (sampaiodiego) [reply packages/rocketchat-blockstack/server/main.js]: actually, this is [set by default](https://docs.meteor.com/api/core.html#Meteor-isDevelopment).. is there a reason to override the default value/logic?

Comment (rodrigok) [reply packages/rocketchat-blockstack/server/main.js]: There isn't, was a trash from the original PR

File: packages/rocketchat-blockstack/server/routes.js (hunk 1 above)
Comment (rodrigok) [reply packages/rocketchat-blockstack/server/routes.js]: @sampaiodiego can you change this?

**Pull request merged**